chainer/functions/theano/theano_function.py

```diff
@@ -0,0 +1,149 @@
+import collections
+
+import numpy
+import six
+
+try:
+    import theano
+    import theano.sandbox.cuda as theano_cuda
+    _available = True
+except ImportError:
+    _available = False
+
+from chainer import cuda
+from chainer import function
+from chainer.utils import type_check
+
+
+def _make_var_tuple(vs):
+    msg = ('inputs and outputs must be a TensorVariable, a list '
+           'of TensorVariable or a tuple of TensorVariable')
+
+    if isinstance(vs, theano.tensor.TensorVariable):
+        return vs,
+    elif isinstance(vs, collections.Iterable):
+        vs = tuple(vs)
```

Comment:
For using `+` operator. `list` is acceptable, but `+` operator with `list` and `tuple` is not permitted.